uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/instances/leave-request.instance.js

```diff
@@ -52,6 +35,32 @@ define([
           }.bind(this));
       }
 
+      /**
+       * Checks if the balance change of the leave request need to be recalculated
+       *
+       * @return {Promise} resolves with {Boolean}
+       */
+      function checkIfBalanceChangeNeedsForceRecalculation () {
+        var originalBalanceAmount, unitName;
+
+        return AbsenceType.all({ id: this.type_id })
```

Comment:
this is fine but it can be refactored like this:

```js
return AbsenceType.all({ id: this.type_id })
  .then(AbsenceType.loadCalculationUnits)
  .then(function (absenceTypes) {
    var unitName = absenceTypes[0].calculation_unit_name;

    return $q.all({
      calculatedBalanceChange: this.calculateBalanceChange(unitName),
      currentBalanceChange: this.getBalanceChangeBreakdown()
    });
  }.bind(this))
  .then(function (results) {
    return +results.currentBalanceChange !== +results.calculatedBalanceChange.amount;
   });
```

Reply:
Beautiful. Done.